### PR TITLE
Basic native support of MSAA for render targets

### PR DESCRIPTION
**Pull request by bghgary**
This change covers the basic case of passing in a sample count for render targets. It does not handle changing the sample count after initialization or multitargets.

This is just the JS changes. Additional native changes are needed for MSAA to work with render targets, but it does not require a protocol version update.

See https://forum.babylonjs.com/t/nativeengine-render-target-texture-not-working-with-msaa/42329.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14055/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14055/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14055/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14055/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14055/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14055/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14055/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14055/merge#BCU1XR#0

File: packages/dev/core/src/Engines/nativeEngine.ts [hunk 1]
```diff
@@ -1864,7 +1864,7 @@ export class NativeEngine extends Engine {
         const width = (<{ width: number; height: number; layers?: number }>size).width || <number>size;
         const height = (<{ width: number; height: number; layers?: number }>size).height || <number>size;
 
-        const framebuffer = this._engine.createFrameBuffer(texture._hardwareTexture!.underlyingResource, width, height, true, true);
+        const framebuffer = this._engine.createFrameBuffer(texture._hardwareTexture!.underlyingResource, width, height, true, true, 1);
```
Comment (Popov72) [inline packages/dev/core/src/Engines/nativeEngine.ts:1867]: Why not using `options.samples || 1` instead of `1`?

**Review by Popov72 — APPROVED** [review]

**Review by sebavan — APPROVED** [review]
LGTM

File: packages/dev/core/src/Engines/nativeEngine.ts (hunk 1 above)
Comment (bghgary) [reply packages/dev/core/src/Engines/nativeEngine.ts:1867]: Hmm, we have a TODO above to handle options, so I ignored the options. Let me see if it'll work.

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14055/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14055/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Pull request merged**